libspawner/src/posix/securerunner.cpp

```diff
@@ -297,6 +298,14 @@ void *secure_runner::check_limits_proc(void *monitor_param) {
             }
         }
 
+        if (self->check_restriction(restriction_memory_limit) &&
```

Comment:
One should expand procfs _stat_ file parser to extract second field (executable name) to make sure you are not judging a forked Spawner process.

Mikhail, look for proc(5) man page for details.